js/woo.js

```diff
@@ -0,0 +1,1469 @@
+        const orderId = this.safeInteger (order, 'order_id');
+        const clientOrderId = this.safeTimestamp (order, 'client_order_id'); // Somehow, this always returns 0 for limit order
+        if (market === undefined) {
+            const marketId = this.safeString (order, 'symbol');
+            market = this.safeMarket (marketId, market);
+        }
+        const symbol = market['symbol'];
+        const price = this.safeString2 (order, 'order_price', 'price');
+        const quantity = this.safeString2 (order, 'order_quantity', 'quantity'); // This is base amount
+        const amount = this.safeString2 (order, 'order_amount', 'amount'); // This is quote amount
+        const orderType = this.safeStringLower2 (order, 'order_type', 'type');
+        const status = this.safeValue (order, 'status');
+        const side = this.safeStringLower2 (order, 'side');
+        const filled = this.safeValue (order, 'executed');
+        const remaining = Precise.stringSub (amount, filled);
+        const fee = this.safeValue (order, 'total_fee');
+        const feeCurrency = this.safeString (order, 'fee_asset');
+        const transactions = this.safeValue (order, 'Transactions');
+        return this.safeOrder ({
+            'id': orderId,
+            'clientOrderId': clientOrderId,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'lastTradeTimestamp': undefined,
+            'status': this.parseOrderStatus (status),
+            'symbol': symbol,
+            'type': orderType,
+            'timeInForce': undefined,
+            'postOnly': undefined, // TO_DO
+            'side': side,
+            'price': price,
+            'stopPrice': undefined,
+            'average': undefined,
+            'amount': quantity, // TO_DO
+            'filled': filled,
+            'remaining': remaining, // TO_DO
+            'cost': amount, // TO_DO
+            'trades': transactions,
+            'fee': {
+                'cost': fee,
+                'currency': feeCurrency,
+            },
+            'info': order,
+        }, market);
+    }
+
+    parseOrderStatus (status) {
+        if (status !== undefined) {
+            const statuses = {
+                'NEW': 'open',
+                'FILLED': 'closed',
+                'CANCEL_SENT': 'canceled',
+                'CANCEL_ALL_SENT': 'canceled',
+            };
+            return this.safeString (statuses, status, status);
+        }
+        return status;
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        if (limit !== undefined) {
+            if (limit > 1000) {
+                throw new BadRequest (this.id + ' fetchOHLCV() limit argument cannot exceed 1000');
+            }
```

Comment:
Let's change lines 830-833 like this:

```JavaScript
if (limit !== undefined) {
    limit = Math.min (limit, 1000);
```

Reply:
it that change the unified approach, to silently fix (trim down or whatever) user inputed parameter? because I remind other exchagnes, where you have to input explicitly i.e. `20,50` or whatever number, and if user doesn't enter those numbers, it throws exception. (so, maybe in those cases too, CCXT should trim down to nearest number? i.e. if user inputs 39, then it should rounddown to 20, if user inputs 83, then rounddown to 50... so, no exceptions should be thrown?)

Reply:
> maybe in those cases too, CCXT should trim down to nearest number?

I would not truncate the limit in those cases at this point, we can add that later if necessary.